File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/permissionchecker/PermissionCheckerService.java
package com.nivasafinance.features.rolemanagement.permissionchecker;

import java.util.List;

import com.nivasafinance.features.rolemanagement.permission.dto.PermissionBatchCheckResponse;
import com.nivasafinance.features.rolemanagement.permission.dto.PermissionCheckResponse;

public interface PermissionCheckerService {
    boolean checkPermissionForUser(
            String username,
            String permissionName
    );

    PermissionCheckResponse checkPermission(String permissionName);

    PermissionBatchCheckResponse checkPermissionsBatch(List<String> permissionNames);
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/permissionchecker/PermissionCheckerService.kt
package com.nivasafinance.features.rolemanagement.permissionchecker

import com.nivasafinance.features.rolemanagement.enums.ActionEnum
import com.nivasafinance.features.rolemanagement.enums.ModuleEnum
import com.nivasafinance.features.rolemanagement.enums.OperationsEnum

interface PermissionCheckerService {
    fun checkPermissionForUser(
        username: String,
        action: ActionEnum,
        module: ModuleEnum,
        operation: OperationsEnum
    ): Boolean
}
